src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,184 @@
+                    } else if (input.contains("deadline")) {
+                        task = parser.handleDeadline(input);
+                    } else if (input.contains("event")) {
+                        task = parser.handleEvent(input);
+                    } else {
+                        throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+                    }
+
+                    System.out.println(taskList.add(task));
+                    storage.save();
+                } catch (DukeException e) {
+                    System.out.println("    ____________________________________________________________");
+                    System.out.println("     " + e.getMessage());
+                    System.out.println("    ____________________________________________________________");
+                    System.out.println();
+                }
+            }
+        }
+        sc.close();
+    }
+
+    /**
+     * Reads input from the user through the GUI and parses the input.
+     * For use with Duke GUI.
+     * @param input User input
+     * @return Output to be displayed to user
+     */
+    public String readCommand(String input) {
+
+        if (input.equals("bye")) {
+            return "Have a nice day! :-)";
+        } else if (input.equals("list")) {
+            return taskList.listOut();
+        } else if (input.equals("help")) {
+            PopUpBox.display("Help menu", HELP_TEXT);
+            return "Here is the list of commands:";
+        } else if (input.contains("find")) {
+            return taskList.find(input);
+        } else if (input.contains("done")) {
+            try {
+                String output = taskList.done(input);
+                storage.save();
+                return output;
+            } catch (DukeException e) {
+                return e.getMessage();
+            }
+        } else if (input.contains("delete")) {
+            try {
+                String output = taskList.delete(input);
+                storage.save();
+                return output;
+            } catch (DukeException e) {
+                return e.getMessage();
+            }
+        } else {
+            try {
+                Task task;
+                if (input.contains("todo")) {
+                    task = parser.handleToDo(input);
+                } else if (input.contains("deadline")) {
+                    task = parser.handleDeadline(input);
+                } else if (input.contains("event")) {
+                    task = parser.handleEvent(input);
+                } else {
+                    throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+                }
+
+                String output = taskList.add(task);
+                storage.save();
+                return output;
+            } catch (DukeException e) {
+                return e.getMessage();
+            }
+        }
+    }
+
+    /**
+     * Greeting used by Duke.

```

Comment:
Similarly, I think it would be better to write "Greets the user" for proper javaDoc.